index.html

```diff
@@ -393,6 +397,47 @@
         }
       }
 
+      // Circle & pin functionality:
+
+      function find_circle(x, y) {
+        var drawing = document.getElementById("drawing");
+
+        for (let element of drawing.children) {
+          if (element.tagName === "circle" && element.getAttribute("cx") === x && element.getAttribute("cy") === y && element.getAttribute("r") === "10") {
```

Comment:
Note that this loop is looping throught the svg (drawing), selecting all circles. This means both dots (`<circle>` with fill) and circles (`<circle>` without fill).